overlord/devicestate/devicemgr.go

```diff
@@ -267,6 +270,11 @@ func (m *DeviceManager) ensureOperational() error {
 
 	perfTimings := timings.New(map[string]string{"ensure": "become-operational"})
 
+	if m.operatingMode == "install" {
```

Comment:
Do we have a test for this? Iirc one was added in the bigger PR, no?

Reply:
In fact `TestDeviceRegistrationNotInInstallMode()` is already in `devicestate_serial_test.go`, as is the code to prevent registration in install mode already in `ensureOperational()`. Removing the duplicate test.